File: app/src/main/java/com/github/goldy1992/mp3player/client/views/adapters/RecyclerViewAdapters.java
package com.github.goldy1992.mp3player.client.views.adapters;

import androidx.annotation.Nullable;

import com.github.goldy1992.mp3player.commons.MediaItemType;
import com.github.goldy1992.mp3player.dagger.scopes.ComponentScope;

import javax.inject.Inject;

@ComponentScope
public class RecyclerViewAdapters {

    private final MySongViewAdapter mySongViewAdapter;
    private final MyFolderViewAdapter myFolderViewAdapter;

    @Inject
    public RecyclerViewAdapters(MySongViewAdapter mySongViewAdapter,
                                MyFolderViewAdapter myFolderViewAdapter) {
        this.myFolderViewAdapter = myFolderViewAdapter;
        this.mySongViewAdapter = mySongViewAdapter;
    }

    @Nullable
    public MyGenericRecycleViewAdapter getAdapter(MediaItemType mediaItemType) {
        switch (mediaItemType) {
            case SONGS:
            case FOLDER: return mySongViewAdapter;
            case FOLDERS: return myFolderViewAdapter;
            default: return null;
        }
    }
}

File: app/src/main/java/com/github/goldy1992/mp3player/client/views/adapters/RecyclerViewAdapters.kt
package com.github.goldy1992.mp3player.client.views.adapters

import com.github.goldy1992.mp3player.commons.MediaItemType
import com.github.goldy1992.mp3player.dagger.scopes.ComponentScope
import javax.inject.Inject

@ComponentScope
class RecyclerViewAdapters @Inject constructor(private val mySongViewAdapter: MySongViewAdapter,
                                               private val myFolderViewAdapter: MyFolderViewAdapter) {
    fun getAdapter(mediaItemType: MediaItemType?): MyGenericRecycleViewAdapter? {
        return when (mediaItemType) {
            MediaItemType.SONGS, MediaItemType.FOLDER -> mySongViewAdapter
            MediaItemType.FOLDERS -> myFolderViewAdapter
            else -> null
        }
    }

}